### PR TITLE
Fixes 4110,4209: set gpg & modular hotfixes in candlepin

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

Adds support for gpg URL (On Content) & Modular hotfixes (as an override) in candlepin
Adds an update repository task to update these settings

## Testing steps

1.  Register a system following https://github.com/content-services/content-sources-backend/blob/main/docs/register_client.md
3. Create a custom repository with no GPG key and module filtering enabled (defaults)
4. Create a content template with the repository
5. associate the system to the template
6. On the client,  run   `subscription-manager refresh && subscription-manager repo-override && yum repolist`  check the /etc/yum.repos.d/redhat.repo file.  Verify that gpgcheck=0  and module_hotfixes=1 is not present.
7. Update the repository, add a GPG Key, toggle modularity filtering
8. On the client, run   `subscription-manager refresh && subscription-manager repo-override && yum repolist`  check the /etc/yum.repos.d/redhat.repo file.  Verify that gpgcheck=1, gpgkey=$URL  and module_hotfixes=1 is present.
9. Undo 7, and redo 8, and verify the settings reverted back to their normal behavior.
## Checklist

- [ ] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4110

File: pkg/candlepin_client/content.go [hunk 1]
```diff
@@ -70,3 +71,38 @@ func (c *cpClientImpl) CreateContent(ctx context.Context, ownerKey string, conte
...
+
+	content, httpResp, err := client.OwnerContentAPI.GetContentById(ctx, OwnerKey(orgID), GetContentID(repoConfigUUID)).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			return nil, nil
+		}
+		return nil, errorWithResponseBody("couldn't update content", httpResp, err)
```
Comment (rverdile) [inline pkg/candlepin_client/content.go:105]:
```suggestion
		return nil, errorWithResponseBody("couldn't fetch content", httpResp, err)
```

File: pkg/tasks/update_repository.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		return err
+	}
+	if content == nil {
+		// content hasn't been created yet, so no need to update
+		return nil
+	}
+
+	if err := ur.UpdateGPGKey(*content); err != nil {
+		if err != nil {
+			return fmt.Errorf("could not udpate GPG Key %w", err)
```
Comment (rverdile) [inline pkg/tasks/update_repository.go:83]:
```suggestion
			return fmt.Errorf("could not udpate GPG Key: %w", err)
```

File: pkg/tasks/update_repository.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+		}
+	}
+	if err := ur.UpdateContentOverrides(); err != nil {
+		if err != nil {
+			return fmt.Errorf("could not update environment overrides %w", err)
```
Comment (rverdile) [inline pkg/tasks/update_repository.go:88]:
```suggestion
			return fmt.Errorf("could not update environment overrides: %w", err)
```

File: pkg/tasks/update_repository.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+func (ur *UpdateRepository) UpdateGPGKey(content caliri.ContentDTO) error {
+	expected := GenContentDto(ur.repoConfig)
+	if content.GpgUrl == nil && expected.GpgUrl == nil {
+		return nil
+	} else if content.GpgUrl != nil && expected.GpgUrl != nil && content.GpgUrl == expected.GpgUrl {
+		return nil
+	} else {
+		err := ur.cpClient.UpdateContent(ur.ctx, ur.orgID, ur.repoConfig.UUID, expected)
+		if err != nil {
+			return fmt.Errorf("could not repository for gpg key update %w", err)
```
Comment (rverdile) [inline pkg/tasks/update_repository.go:103]: > could not repository for gpg key update

there's a word missing here! i can't figure out what you meant :)

**Comment by jlsherrill** [comment]
oh and @swadeley i don't think there's anything to test here yet, but will be in the future when candlepin integration is turned on

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4209

File: pkg/config/config.go [hunk 5]
```diff
@@ -237,6 +238,7 @@ func setDefaults(v *viper.Viper) {
 	v.SetDefault("options.enable_notifications", false)
 	v.SetDefault("options.template_event_topic", "platform.content-sources.template")
 	v.SetDefault("options.repository_import_filter", "")
+	v.SetDefault("options.external_url", "http://pulp.content:8000")
```
Comment (jlsherrill) [inline pkg/config/config.go:241]: oh i realized i need to add this to the deployment.yaml  doing that now

**Review by rverdile — APPROVED** [review]

**Comment by rverdile** [comment]
Is QE needed now that 4209 is included? The automation seems confused

**Comment by jlsherrill** [comment]
oh yeah, it probably is!  let me change it

**Comment by swadeley** [comment]
Hi @jlsherrill 

Re: HMS-4209 API - Modify template API to accept a list of repo_uuids

I think we need an apidocs update:

```
In [12]: app.content_sources.rest_client.templates_api.list_templates.attribute_map
Out[12]: 
{'offset': 'offset',
 'limit': 'limit',
 'version': 'version',
 'arch': 'arch',
 'name': 'name',
 'sort_by': 'sort_by'}

```

**Comment by swadeley** [comment]
Hi

Looks good:

```
In [8]: app.content_sources.rest_client.templates_api.create_template(template_dict)
2024-06-07 20:49:41.180 [    INFO] [iqe.base.rest_client] REST: METHOD=POST, request_id=8<>, params=[]
Out[8]: 
{'arch': 'x86_64',
 'date': '0001-01-01T00:00:00Z',
 'description': 'my 2nd template',
 'name': 'test2 el8',
 'org_id': '3340851',
 'repository_uuids': ['c29491cb-fde9-4d3e-8948-97d8bdc09cd8'],
 'rhsm_environment_id': '2a6d25116b584dc3bf62a83989d2f76c',
 'uuid': '2a6d2511-6b58-4dc3-bf62-a83989d2f76c',
 'version': '8'}

In [9]: template_dict = dict(name="test1 el8",repository_uuids=[created_repo1.uuid],description="my 1st template",arch="x86_64",version="8")

In [10]: app.content_sources.rest_client.templates_api.create_template(template_dict)
2024-06-07 20:49:55.331 [    INFO] [iqe.base.rest_client] REST: METHOD=POST, request_id=<>, params=[]
Out[10]: 
{'arch': 'x86_64',
 'date': '0001-01-01T00:00:00Z',
 'description': 'my 1st template',
 'name': 'test1 el8',
 'org_id': '3340851',
 'repository_uuids': ['663b20a8-bfe8-49b1-9290-e1dba054294c'],
 'rhsm_environment_id': '8cb56d03502d4add8127cfca3a624cc5',
 'uuid': '8cb56d03-502d-4add-8127-cfca3a624cc5',
 'version': '8'}

In [11]:
```

at last:

```
In [21]: app.content_sources.rest_client.templates_api.list_templates(repository_uuids="'663b20a8-bfe8-49b1-9290-e1dba054294c','c29491cb-fde9-4d3e-8948-97d8bdc09cd8'")
2024-06-07 21:00:51.481 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=<>, params=[('repository_uuids', "'663b20a8-bfe8-49b1-9290-e1dba054294c','c29491cb-fde9-4d3e-8948-97d8bdc09cd8'")]
Out[21]: 
{'data': [{'arch': 'x86_64',
           'date': '0001-01-01T00:00:00Z',
           'description': 'my 1st template',
           'name': 'test1 el8',
           'org_id': '3340851',
           'repository_uuids': ['663b20a8-bfe8-49b1-9290-e1dba054294c'],
           'rhsm_environment_id': '8cb56d03502d4add8127cfca3a624cc5',
           'uuid': '8cb56d03-502d-4add-8127-cfca3a624cc5',
           'version': '8'},
          {'arch': 'x86_64',
           'date': '0001-01-01T00:00:00Z',
           'description': 'my 2nd template',
           'name': 'test2 el8',
           'org_id': '3340851',
           'repository_uuids': ['c29491cb-fde9-4d3e-8948-97d8bdc09cd8'],
           'rhsm_environment_id': '2a6d25116b584dc3bf62a83989d2f76c',
           'uuid': '2a6d2511-6b58-4dc3-bf62-a83989d2f76c',
           'version': '8'}],
 'links': {'first': "/api/content-sources/v1/templates/?limit=100&offset=0&repository_uuids='663b20a8-bfe8-49b1-9290-e1dba054294c','c29491cb-fde9-4d3e-8948-97d8bdc09cd8'",
           'last': "/api/content-sources/v1/templates/?limit=100&offset=0&repository_uuids='663b20a8-bfe8-49b1-9290-e1dba054294c','c29491cb-fde9-4d3e-8948-97d8bdc09cd8'"},
 'meta': {'count': 2, 'limit': 100, 'offset': 0}}

In [22]: 
```

**Pull request merged**